pkgs/development/python-modules/myst-parser/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, flit-core
+, docutils
+, mdit-py-plugins
+, pyyaml
+, sphinx
+, typing-extensions
+  # , beautifulsoup4
+  # , coverage
+  # , pytest-cov
+  # , pytest-param-files
+  # , pytest-regressions
+  # , sphinx-pytest
+  # , pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "myst-parser";
+  version = "0.18.0";
+
+  src = fetchFromGitHub {
+    owner = "executablebooks";
+    repo = pname;
+    rev = "v" + version;
+    sha256 = "sha256-GEtrC7o5YnkuvBfQQfhG5P74QMiHz63Fdh1cC/r5CF0=";
+  };
+
+  format = "flit";
+
+  nativeBuildInputs = [ flit-core ];
+
+  propagatedBuildInputs = [
+    docutils
+    mdit-py-plugins
+    pyyaml
+    sphinx
+    typing-extensions
+  ];
```

Comment:
There could be more runtime dependencies: [`jinja2`](https://github.com/executablebooks/MyST-Parser/blob/v0.18.0/pyproject.toml#L38) and [`markdown-it-py`](https://github.com/executablebooks/MyST-Parser/blob/v0.18.0/pyproject.toml#L39).